### PR TITLE
Ignore redundant copies of parts after move and restart

### PR DESCRIPTION
**Pull request by excitoon**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):

Ignore redundant copies of parts after move and restart. Progress on #7660.

--
Progress on #7660

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
All `test_multiple_disks` tests has failed.

File: dbms/src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -760,6 +766,13 @@ void MergeTreeData::loadDataParts(bool skip_sanity_checks)
             if (startsWith(it.name(), "tmp"))
                 continue;
 
+            Poco::Path marker_path(it.path(), DELETE_ON_DESTROY_MARKER_PATH);
```
Comment (alesapin) [inline dbms/src/Storages/MergeTree/MergeTreeData.cpp]: This operation extremely dangerous. Need logging.

File: dbms/src/Storages/MergeTree/MergeTreeData.cpp [hunk 2]
```diff
@@ -2528,6 +2541,17 @@ void MergeTreeData::swapActivePart(MergeTreeData::DataPartPtr part_copy)
...
             modifyPartState(part_it, DataPartState::Committed);
+
+            Poco::Path marker_path(Poco::Path(original_active_part->getFullPath()), DELETE_ON_DESTROY_MARKER_PATH);
+            try
+            {
+                Poco::File(marker_path).createFile();
+            }
+            catch (...)
+            {
+                LOG_WARNING(log, "Exception has occurred while creating DeleteOnDestroy marker: '"
```
Comment (alesapin) [inline dbms/src/Storages/MergeTree/MergeTreeData.cpp]: Need to log information about exception.

Comment (excitoon) [reply dbms/src/Storages/MergeTree/MergeTreeData.cpp]: +

File: dbms/src/Storages/MergeTree/MergeTreeData.cpp (hunk 1 above)
Comment (excitoon) [reply dbms/src/Storages/MergeTree/MergeTreeData.cpp]: Added logging and switched to detaching instead of removing.

**Comment by excitoon** [comment]
I need to add a test here.

**Comment by excitoon** [comment]
I would like to add a test for delayed merge as well.

**Comment by excitoon** [comment]
Safe to merge from my side.

**Pull request merged**